snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotificationHandler.java

```diff
@@ -229,49 +232,29 @@ public void showSampleNotification() {
                 .setSubText("Snapyr: Subtext")
                 .setContentText("Snapyr: Content text")
                 .setAutoCancel(true);
-
-        Random r = new Random();
-        Intent baseIntent = getLaunchIntent();
-
-        if (baseIntent == null) {
-            Log.e("SnapyrPaul", "Base intent was null, can't continue.");
-            return;
-        }
-
-        //        Intent buttonIntent = new Intent(applicationContext, applicationContext.activ);
-        //        buttonIntent.putExtra(ACTION_ID_KEY, actionButton.getString(ACTION_ID_KEY));
-        //        buttonIntent.putExtra(NOTIF_DEEP_LINK_KEY, buttonDeepLinkUrl);
-        //        buttonIntent.putExtra(ACTION_TOKEN_KEY, buttonToken);
-        //        buttonIntent.putExtra(INTERACTION_KEY, INTERACTION_TYPE.ACTION_BUTTON_PRESS);
-        //        buttonIntent.putExtra("notificationId", 123);
-
-        // Clone base intent retrieved from package, above.
-        Intent buttonIntent = new Intent(baseIntent);
-
-        PendingIntent buttonAction =
-                PendingIntent.getActivity(applicationContext, r.nextInt(), buttonIntent, 0);
-        builder.addAction(R.drawable.ic_snapyr_logo_only, "1", buttonAction);
-
-        buttonIntent = new Intent(baseIntent);
-        buttonAction = PendingIntent.getActivity(applicationContext, r.nextInt(), buttonIntent, 0);
-        builder.addAction(
-                R.drawable.ic_snapyr_logo_only, "2Supercalifragilisticexpalidocious", buttonAction);
-
-        buttonIntent = new Intent(baseIntent);
-        buttonAction = PendingIntent.getActivity(applicationContext, r.nextInt(), buttonIntent, 0);
-        builder.addAction(R.drawable.ic_snapyr_logo_only, "a", buttonAction);
+        int notificationId = ++nextMessageId;
+        createActionButton(builder, notificationId, "button_one", "button_one",
+                Uri.parse("some_url"), INTERACTION_TYPE.NOTIFICATION_PRESS);
+        createActionButton(builder, notificationId, "button_two", "button_two",
+                Uri.parse("other_url"), INTERACTION_TYPE.NOTIFICATION_PRESS);
 
         builder.setContentIntent(this.getDeepLinkIntent("test%20builtin"));
         Notification notification = builder.build();
         notificationMgr.notify(nextMessageId++, notification);
     }
 
+    SnapyrNotificationListener listener = new SnapyrNotificationListener();
+
     public PendingIntent getDeepLinkIntent(String extra) {
         Intent deepLinkIntent =
                 new Intent(Intent.ACTION_MAIN, Uri.parse("snapyrsample://test/Alice/" + extra));
         return PendingIntent.getActivity(applicationContext, 0, deepLinkIntent, 0);
     }
 
+    private class NotificationHandler extends Activity{
+
```

Comment:
This is going to be used somewhere in the future? Like for handling notification actions?

Reply:
Nope, that's dead code. Good catch